### PR TITLE
Fixes #13 - Add Continuous Integration

### LINKED ISSUE #13: Add Continuous Integration and related plugins/services
Please setup CircleCI, Code Climate, and access-lint accessibility integration for this repository.

### PR DESCRIPTION
**Pull request by VivianChu**

### CODE REVIEW

**Comment by VivianChu** [comment]
@ucsdlib/developers - Please review, comment, etc... Thanks

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+general:
+  artifacts:
+    - "tmp"
+
+machine:
+  node:
+    version: 6.1.0
+  ruby:
+    version:
+      2.2.3
```
Comment (mcritchlow) [inline circle.yml]: Can we change to 2.4?

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    - wget http://projects.iq.harvard.edu/files/fits/files/fits-1.0.5.zip -O fits-1.0.5.zip
+    - unzip fits-1.0.5.zip -d /tmp/
+    - sudo chmod 755 /tmp/fits-1.0.5/fits.sh
+    - sudo ln -s /tmp/fits-1.0.5/fits.sh /bin/fits.sh
+  override:
+    - bundle check --path=vendor/bundle || bundle install --path=vendor/bundle --jobs=4 --retry=3
+  post:
+    - bundle exec rake hydra:test_server:
+        background: true  
+    - sleep 10 && wget --waitretry=5 --retry-connrefused -v http://localhost:8986/rest
```
Comment (mcritchlow) [inline circle.yml]: It looks like this checks for Fedora to be started and assuming Solr succeeds. That's probably true the majority of the time, but in theory starting up Solr could fail as well. Also if for some reason CircleCI is particularly slow, I wonder if we might go past the 5 retries?

What if we created a little script in our horton bin directory that we can call like:
`bin/test_server_wait`

And then populate that script with checks for Fedora and Solr. Something like the little script below. I'm not sure what the wget equivalent of this is, I'm just a bit more familiar with curl.

@jhriv - may actually have a cleaner way of doing this we could try also

```
#!/usr/bin/env sh

printf 'Starting Solr.'
until $(curl --output /dev/null --silent --head --fail http://localhost:8985/solr); do
    printf '.'
    sleep 1
done

printf 'Starting Fedora.'
until $(curl --output /dev/null --silent --head --fail http://localhost:8986/rest); do
    printf '.'
    sleep 1
done
```

**Comment by jhriv** [comment]
Besides a curl test, a netstat or ss call can be made to see if anything is listening on the port, but the service may not be ready as soon as the port is open.

ss -tl '( sport = :8985 )' | grep-q LISTEN
netstat -nl | grep -q 8985

**Comment by VivianChu** [comment]
@mcritchlow - The script has been added and circleci config has been updated.  It's working fine.  However, bundle install failed with ruby version 2.4.

**Comment by mcritchlow** [comment]
@VivianChu very strange about 2.4 not working, but we can always circle back to updating Ruby as a separate task.

**Comment by mcritchlow** [comment]
@lsitu or @hweng any concerns?

**Comment by lsitu** [comment]
👍

**Comment by lsitu** [comment]
@mcritchlow I upgrade ruby to 2.4.0 and it works in my local environment.

**Comment by VivianChu** [comment]
@mcritchlow - Here's the error message in here https://circleci.com/gh/ucsdlib/horton/66
make failed, exit code 2

Gem files will remain installed in
/home/ubuntu/horton/vendor/bundle/ruby/2.4.0/gems/json-1.8.3 for inspection.
Results logged to
/home/ubuntu/horton/vendor/bundle/ruby/2.4.0/extensions/x86_64-linux/2.4.0-static/json-1.8.3/gem_make.out

An error occurred while installing json (1.8.3), and Bundler cannot
continue.

**Comment by mcritchlow** [comment]
@VivianChu @lsitu Interesting. It looks like there was/is an issue with that version of the JSON gem and Ruby/Rails compatibility https://github.com/rails/rails/issues/27450#issuecomment-269500870

**Comment by lsitu** [comment]
@mcritchlow @VivianChu Yes, it's strange. Maybe we can try 2.3.1 for now or just circle it back later?

**Comment by mcritchlow** [comment]
How about we get this merged so we can then get the other PR's merged, and then I'll setup another ticket we can work on next Sprint for getting Ruby updated?

**Comment by lsitu** [comment]
@mcritchlow Yes. It sounds good and I'll merge it now.

**Pull request merged**